flocker/node/_docker.py

```diff
@@ -508,6 +508,9 @@ def _remove():
                     break
 
             try:
+                while (self._client.inspect_container(container_name)
+                        ['State']['Running']):
+                    sleep(0.1)
```

Comment:
This fix seems like a work-around for a bug in either the Python Docker client or Docker itself.  

  * There should be a link to an upstream bug report about the misbehavior.
  * We should log iterations in this loop (perhaps including information about the container state - ideally, the complete output of `inspect_container`, I think) so we can see how often this happens, how long we end up having to wait, and so that if someone ever ends up stuck in an infinite loop here, they can see what's going on.
  * I see that Docker logs information about its attempts to stop containers.  It logs `Failed to send SIGTERM to the process, force killing` and `Container %v failed to exit within %d seconds of SIGTERM - using the force`.  Then ... it calls `WaitStop` with an indefinite timeout and ignores the return (which may be an error).  Considering these containers do seem to eventually stop that's probably not the case we're hitting (yet :cry:).
  * Since it's still not clear what's going on here, enabling docker-py logging (which I did in another branch for this issue) seems like a good idea.  Perhaps Docker is actually working fine but docker-py is hiding some other failure from us.  Being able to reproduce the failure on demand would be the best way to determine that.  Since it seems we can't do that, logging the HTTP traffic seems like the next best thing.